app/src/main/java/com/randomappsinc/studentpicker/home/MainActivity.java

```diff
@@ -310,4 +323,42 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+
+    private void showRenameDialog(final int position) {
+        new MaterialDialog.Builder(this)
+                .title(R.string.rename_list)
+                .input(getString(R.string.new_list_name), "", (dialog, input) -> {
+                    boolean submitEnabled = !(input.toString().trim().isEmpty() ||
+                            preferencesManager.doesListExist(input.toString().trim()));
+                    dialog.getActionButton(DialogAction.POSITIVE).setEnabled(submitEnabled);
+                })
+                .alwaysCallInputCallback()
+                .negativeText(android.R.string.cancel)
+                .onAny((dialog, which) -> {
+                    if (which == DialogAction.POSITIVE) {
+                        String newListName = dialog.getInputEditText().getText().toString().trim();
+                        dataSource.renameList(nameListsAdapter.getItem(position), newListName);
+                        preferencesManager.renameList(nameListsAdapter.getItem(position), newListName);
+                        nameListsAdapter.getContent().set(position, newListName);
+                        nameListsAdapter.notifyItemChanged(position);
+                    }
+                })
+                .show();
+    }
+
+    private void showDeleteDialog(final int position) {
```

Comment:
Same feedback as above. Let's turn this deletion dialog into another module that the activity contains and coordinates.